script.js

```diff
@@ -1,4 +1,598 @@
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  };
+  return cardDeck;
+};
+
+//shuffled card deck
+var shuffledDeck = shuffleCards(cardDeck);
+
+//keeps track of game mode - starts with pre-game
+var GAME_MODE_PRE_GAME = `pre game`;
+var GAME_MODE_BETTING = `betting`;
+var GAME_MODE_INITIAL = `initial`;
+var GAME_MODE_HIT_OR_STAND = `hit or stand`;
+var GAME_MODE_END = `end turn`;
+var gameMode = GAME_MODE_PRE_GAME;
+
+//pre-game message
+var instructions = `Hello, welcome to Blackjack!<br><br>
+The rules of the game are as follows:<br>
+1. The goal of blackjack is to have a hand that totals higher than the dealer's but doesn't total higher than 21<br>
+2. If your hand totals higher than 21, it is called a bust, which means you are out of the game<br>
+3. Before the game starts, all players place a bet<br>
+4. The dealer will deal 2 face up cards to each player, and a face up card and face down card to himself<br>
+5. Cards 2-10 are scored using their face value, and J Q K are equal to 10, but aces can be 1 or 11<br>
+6. If your 2 face up cards total 21, you win 1.5 times your bet<br>
+7. Otherwise, each player can choose to hit (draw another card) or stand (end turn)<br>
+8. If you bust, the dealer wins your bet<br>
+9. After all players are done, the dealer will hit if their score is less than 17<br>
+10. If the dealer busts, every player still in the game wins 2 times their bet<br>
+11. If dealer does not bust, players who score higher than the dealer win 2 times their bet<br>
+12. Everyone else loses their bet to the dealer
+<br><br>
+Please enter number of players to start.`
+
+//keeps track of number of players
+var numPlayers = 0;
+var createNumPlayers = function(number){
+  numPlayers = Number(number);
+  return `There are ${numPlayers} players in this game.`;
+};
+
+//creates player hands
+var createPlayerHands = function() {
+  var playerHands = [];
+  var playerIndex = 1;
+    while (playerIndex <= numPlayers) {
+      var playerHand = {
+        player: playerIndex,
+        hand:[],
+        bank: 100,
+        bet: 0,
+        score: 0,
+        result: ``
+      };
+      playerHands.push(playerHand);
+      playerIndex += 1;
+    };
+  return playerHands;
+};
+
+//keeps track of dealer and player hands
+var dealerHand = [];
+var playerHands = [];
+
+//keeps track of which player's turn it is - start with player 1
+var playerTurn = 0;
+// var allPlayersHaveGone = false;
+
+//moves to next player's turn until every player has gone
+var nextTurn = function(){
+  playerTurn += 1;
+  if(playerTurn > (numPlayers-1)){
+    playerTurn = 0;
+    // allPlayersHaveGone = true;
+  };
+  return `It's player ${playerHands[playerTurn].player}'s turn.`
+};
+
+//creates a player's bet on their turn
+var createBet = function(input,playerTurn){
+  var bet = Number(input);
+  playerHands[playerTurn].bank -= bet;
+  //checks that bet is not blank and is a number
+  if (Number.isNaN(bet) ||
```

Comment:
if you're using if/return, you don't need to use else as well

```
if(playerHands[playerTurn].bank < 0){
    playerHands[playerTurn].bank +=bet;
    return `Player ${playerHands[playerTurn].player} doesn't have enough in the bank!<br>
    Player ${playerHands[playerTurn].player} bank: ${playerHands[playerTurn].bank}`;
  }

 if (playerHands[playerTurn].player < numPlayers){
    playerHands[playerTurn].bet = bet;
    var playerBetMessage = `Player ${playerHands[playerTurn].player} bet $${playerHands[playerTurn].bet}.`
    var nextTurnMessage = nextTurn();
    return `${playerBetMessage}<br>${nextTurnMessage}` }
```
if you use if /else if, you return at the end of the code block, not at the end of every if statement